contracts/oracle/CollateralizationMemoizer.sol

```diff
@@ -0,0 +1,204 @@
+contract CollateralizationMemoizer is IOracle, CoreRef {
+    using Decimal for Decimal.D256;
+
+    // ----------- Events -----------
+
+    event CachedValueUpdate(
+        address from,
+        uint256 indexed protocolControlledValue,
+        uint256 indexed userCirculatingFei,
+        int256 indexed protocolEquity
+    );
+
+    // ----------- Properties -----------
+
+    /// @notice address of the CollateralizationOracle to memoize
+    address public collateralizationOracle;
+
+    /// @notice cached value of the Protocol Controlled Value
+    uint256 public cachedProtocolControlledValue;
+    /// @notice cached value of the User Circulating FEI
+    uint256 public cachedUserCirculatingFei;
+    /// @notice cached value of the Protocol Equity
+    int256 public cachedProtocolEquity;
+
+    /// @notice timestamp of the last read operation on the CollateralizationOracle
+    uint256 public lastUpdate;
+    /// @notice validity duration of the cached value
+    uint256 public validityDuration;
+    /// @notice deviation threshold to consider cached values outdated, in basis
+    ///         points (base 10_000)
+    uint256 public deviationThreshold;
+
+    // ----------- Constructor -----------
+
+    /// @notice CollateralizationOracle constructor
+    /// @param _core Fei Core for reference
+    constructor(
+        address _core
+    ) CoreRef(_core) {}
+
+    // ----------- IOracle override methods -----------
+    /// @notice update reading of the CollateralizationOracle
+    function update() external override whenNotPaused {
+        // fetch a fresh round of information
+        (
+            uint256 _protocolControlledValue,
+            uint256 _userCirculatingFei,
+            int256 _protocolEquity,
+            bool _validityStatus
+        ) = ICollateralizationOracle(collateralizationOracle).pcvStats();
+
+        // only update if valid
+        require(_validityStatus, "CollateralizationMemoizer: CollateralizationOracle is invalid");
+
+        // Update cached variables
+        _setCache(_protocolControlledValue, _userCirculatingFei, _protocolEquity);
+    }
+
+    // @notice returns true if the cached values are outdated.
+    function isOutdated() external override view returns (bool outdated) {
+        // check if cached value is fresh
+        if (block.timestamp > lastUpdate + validityDuration) {
+            outdated = true;
+        }
+        // check if deviation thresholds are met
+        if (!outdated) {
+            (
+                uint256 _protocolControlledValue,
+                uint256 _userCirculatingFei,
+                , // int256 _protocolEquity,
+                bool _validityStatus
+            ) = ICollateralizationOracle(collateralizationOracle).pcvStats();
+
+            require(_validityStatus, "CollateralizationMemoizer: CollateralizationOracle reading is invalid");
+
+            Decimal.D256 memory _thresholdLow = Decimal.from(10_000 - deviationThreshold).div(10_000);
```

Comment:
maybe add a helper method that returns % diff between two numbers then invalidate if that number exceeds the threshold. Would be simpler logic